src/main/java/pl/gov/coi/pomocua/ads/legalassistances/LegalAssistanceResource.java

```diff
@@ -0,0 +1,36 @@
+package pl.gov.coi.pomocua.ads.legalassistances;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.Optional;
+
+
+@RestController
+@RequestMapping(path = "/api/", produces = "application/json")
+@RequiredArgsConstructor
+public class LegalAssistanceResource {
+
+    private final LegalAssistanceRepository legalAssistanceRepository;
+
+    @GetMapping(value = "legal-assistances")
+    public Page<LegalAssistanceOffer> getLegalAssistances(Pageable pageRequest){
+        return legalAssistanceRepository.findAll(pageRequest);
+    }
+
+    @GetMapping(value = "legal-assistances/{id}")
+    public Optional<LegalAssistanceOffer> getLegalAssistance(@PathVariable Long id){
+        return legalAssistanceRepository.findById(id);
```

Comment:
Here we should use `ResponseEntity.of(repository.findById(...))` (see other resource classes)
If we return an empty optional with the use of that, HTTP 404 would be returned instead of "null" value with HTTP 200

Reply:
Sure, changed :)

Reply:
> Here we should use `ResponseEntity.of(repository.findById(...))` (see other resource classes) If we return an empty optional with the use of that, HTTP 404 would be returned instead of "null" value with HTTP 200

Isn't 404 the proper REST response for an entity not found scenario?

Reply:
Ohhh, yes, sure - I've written it wrong.
I.e. when we return `Optional`, HTTP 200 is returned with "null" body (which is wrong), if we use `ResponseEntity.of(Optional)`, HTTP 404 is return (correct)

Reply:
> If we return an empty optional with the use of that
I think I meant "if we return optional with the use of `ResponseEntity.of(...)`, but it's not a proper sentence I think